cinema/views.py

```diff
@@ -97,10 +111,10 @@
     def get_serializer_class(self):
         if self.action == "list":
             return MovieListSerializer
-
         if self.action == "retrieve":
             return MovieDetailSerializer
-
+        if self.action == "upload_image":
+            return MovieImageSerializer
```

Comment:
Ensure that the `MovieImageSerializer` is correctly used for handling image uploads in the `upload_image` method. This may involve verifying that the serializer is properly defined and handles the image field as expected.